src/cpp/session/modules/rmarkdown/NotebookConditions.cpp

```diff
@@ -34,7 +38,7 @@ namespace {
 SEXP rs_signalNotebookCondition(SEXP condition, SEXP message)
 {
    // extract message (make sure we got one)
-   std::string msg = r::sexp::safeAsString(message, "");
+   std::string msg = r::sexp::asUtf8String(message);
```

Comment:
This is really the heart of the fix; the rest of the PR is just me tidying things up along the way to finding this fix ...